ConditionEngine/src/main/java/com/github/cafdataprocessing/corepolicy/PolicyEngineImpl.java

```diff
@@ -145,7 +145,7 @@ public void execute(Long collectionSequenceId, Document document, Collection<Lon
         for (Long policyTypeId : distinctPolicyTypeIds) {
             Collection<Policy> policiesOfType = policies.stream().filter(p -> p.typeId.equals(policyTypeId)).collect(Collectors.toList());
 
-            Optional<PolicyHandler> policyHandler = policyHandlers.stream().filter(ph -> ph.getPolicyTypeId() == policyTypeId).findFirst();
+            Optional<PolicyHandler> policyHandler = policyHandlers.stream().filter(ph -> ph != null && ph.getPolicyTypeId() == policyTypeId).findFirst();
```

Comment:
Good protection now around the ph being null, just incase somewhere registered duff info.